packages/fluentui/react-northstar/src/components/Tree/TreeItem.tsx

```diff
@@ -238,6 +246,20 @@ export const TreeItem: ComponentWithAs<'div', TreeItemProps> & FluentComponentSt
       _.invoke(predefinedProps, 'onClick', e, titleProps);
     },
   });
+  const handleClick = (e: React.SyntheticEvent) => {
+    handleOnClickOnTreeItemOnly(e);
+    _.invoke(props, 'onClick', e, props);
+  };
+
+  const handleOnClickOnTreeItemOnly = (e: React.SyntheticEvent) => {
+    // onClick listener for mouse click on treeItem DOM only,
+    // which could be triggered by VO+space on selectable tree parent node
+    e.preventDefault();
+    e.stopPropagation();
+    if (e.target === e.currentTarget) {
+      handleSelection(e);
+    }
+  };
```

Comment:
Please why do you need a separate function here?

Reply:
And another thing - what if we call    `e.preventDefault();` and ` e.stopPropagation();` inside the `if()` case?

Reply:
Resolved by this [commit](https://github.com/microsoft/fluentui/pull/15147/commits/2cfd658d6462e95d9f67c079cbb76720ff3a7a7e) 

It's not a separate function anymore (so it can't be called by other code, therefore more readable), and `e.preventDefault()` `e.stopPropagation()` is removed